src/composed/message/types.rs

```diff
@@ -398,7 +400,21 @@ impl Message {
             } => {
                 if let Some(message) = message {
                     match **message {
-                        Message::Literal(ref data) => signature.verify(key, data.data()),
+                        Message::Literal(ref data) => {
+                            match signature.typ() {
+                                SignatureType::Text => {
+                                    // Normalize line endings for "Signature of a canonical text document",
+                                    // see https://www.rfc-editor.org/rfc/rfc4880.html#section-5.2.1
+                                    let normalized: Vec<_> = Normalized::new(
+                                        data.data().iter().cloned(),
+                                        LineBreak::Crlf,
+                                    ).collect();
+
+                                    signature.verify(key, &*normalized)
```

Comment:
The problem here seems to be that `Normalized` doesn't implement `Read` but rather is an iterator of u8s. There are ways to adapt iterator like so: https://crates.io/crates/iter_read but I'm not sure additional dependency is worth it :man_shrugging:

Reply:
Thanks for that pointer! I'd be happy to adjust the PR to use that crate (or another similar mechanism).
My tendency would be to try and enable streaming operation.

Reply:
Maybe wait until @dignifiedquire decides... I'm just throwing random ideas here :)